domain/permission/state/state.go

```diff
@@ -112,140 +88,56 @@ VALUES ($M.uuid, $M.access_type_id, $M.grant_on, $M.grant_to)
 	return userAccess, nil
 }
 
-// findUser finds the user provided exists, hasn't been removed and is not
-// disabled. Return data needed to fill in corePermission.UserAccess.
-func findUser(
-	ctx context.Context,
-	tx *sqlair.TX,
-	userName string,
-) (User, error) {
-	var result User
-
-	getUserQuery := `
-SELECT (u.uuid, u.name, u.display_name, u.created_at, u.disabled) AS (&User.*),
-       creator.name AS &User.created_by_name
-FROM   v_user_auth u
-    LEFT JOIN user AS creator
-    ON        u.created_by_uuid = creator.uuid
-WHERE  u.removed = false AND u.name = $M.name`
-
-	selectUserStmt, err := sqlair.Prepare(getUserQuery, User{}, sqlair.M{})
-	if err != nil {
-		return result, errors.Annotate(err, "preparing select getUser query")
-	}
-
-	err = tx.Query(ctx, selectUserStmt, sqlair.M{"name": userName}).Get(&result)
-	if err != nil && errors.Is(err, sql.ErrNoRows) {
-		return result, errors.Annotatef(usererrors.NotFound, "%q", userName)
-	} else if err != nil {
-		return result, errors.Annotatef(err, "getting user with name %q", userName)
-	}
-	if result.Disabled {
-		return result, errors.Annotatef(usererrors.AuthenticationDisabled, "%q", userName)
-	}
-	return result, nil
+// AddUserPermissionArgs is a specification for adding a user permission.
+type AddUserPermissionArgs struct {
```

Comment:
belongs in the types.go file, or maybe better the domain/permission/types.go file?

Reply:
I'll discuss it with others tomorrow. I'm happy for this to land in the mean time and circle back around if required to fix all the state packages (we're doing it in other places as well).